File: dagger2-study/src/main/java/com/asura/dagger2_study/MainActivity.java
package com.asura.dagger2_study;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;

import com.asura.dagger2_study.step0.Step0Activity;
import com.asura.dagger2_study.step0.Student;
import com.asura.dagger2_study.step1.Step1Activity;

import javax.inject.Inject;

/**
 * Dagger2 学习
 * <p>
 * 简书链接：https://www.jianshu.com/p/b40bcd1a9ec9
 */
public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    private Button btnStep0, btnStep1, btnStep2;
    @Inject
    Student mStudent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        btnStep0 = findViewById(R.id.btn_step0);
        btnStep1 = findViewById(R.id.btn_step1);
        btnStep2 = findViewById(R.id.btn_step2);
        btnStep0.setOnClickListener(this);
        btnStep1.setOnClickListener(this);
        btnStep2.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        Intent intent = new Intent();
        switch (v.getId()) {
            case R.id.btn_step0:
                intent.setClass(MainActivity.this, Step0Activity.class);
                break;
            case R.id.btn_step1:
                intent.setClass(MainActivity.this, Step1Activity.class);
                break;
            case R.id.btn_step2:
                break;
            default:
        }
        startActivity(intent);
    }
}

File: customview-study/src/main/java/com/asura/customview_study/MainActivity.kt
package com.asura.customview_study

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity(), View.OnClickListener {
    override fun onClick(v: View?) {
        var intent: Intent = Intent()
        when (v) {
            btn_drawBoard -> {
                intent.setClass(this@MainActivity, DrawBoardActivity::class.java)
            }
            btn_constraintLayout -> {
                intent.setClass(this@MainActivity, ConstraintLayoutActivity::class.java)
            }
            btn_constraint_transition -> {
                intent.setClass(this@MainActivity, ConstraintLayoutTransitionActivity::class.java)
            }
        }
        startActivity(intent)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btn_drawBoard.setOnClickListener(this)
        btn_constraintLayout.setOnClickListener(this)
        btn_constraint_transition.setOnClickListener(this)
    }
}
